docs/pip.md

```diff
@@ -0,0 +1,163 @@
+
+```ini
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile --output-file=requirements.txt requirements.in
+#
+certifi==2020.6.20        # via requests
+chardet==3.0.4            # via requests
+git+https://github.com/containerbuildsystem/dockerfile-parse  # via -r requirements.in
+idna==2.10                # via requests
+https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip  # via -r requirements.in
+requests==2.24.0          # via -r requirements.in
+ruamel.yaml.clib==0.2.2   # via ruamel.yaml
+ruamel.yaml==0.16.12      # via operator-manifest
+six==1.15.0               # via dockerfile-parse
+urllib3==1.25.10          # via requests
+```
+
+As you can see, `pip-compile` gathered all the recursive dependencies and pinned al PyPI packages.
+It did not pin external dependencies, the mechanism for doing so is specific to Cachito. You can pin
+these beforehand in the `requirements.in` file, but if any of the recursive dependencies are
+external, you may need to edit the generated file anyway.
+
+### Explicit package names
+
+Cachito needs to know the package name of all your dependencies. For PyPI dependencies, this is
+trivial, as the name is already present in the requirements file. For external dependencies,
+Resolving the name may require executing the `setup.py` file. Cachito does have a mechanism for
+extracting package metadata from `setup.py`, but it is very limited. That is why, for external
+dependencies, you will need to explicitly specify package names using one of the mechanisms that
+pip supports.
+
+1. use '@': `dockerfile-parse @ git+https://github.com/containerbuildsystem/dockerfile-parse`
+2. use "#egg": `git+https://github.com/containerbuildsystem/dockerfile-parse#egg=dockerfile-parse`
+
+Note that `pip-compile` seems to ignore the `@` mechanism, so using `#egg` may be preferrable.
+
+### Hash checking
+
+In general, Cachito handles hash checking the same way that pip does. If `--require-hashes` is
+present in the requirements file, or if any dependency uses the `--hash` option, Cachito will
+require that all dependencies specify a hash and will check that the hashes are valid.
+
+For HTTP(S) dependencies, Cachito will always require a hash and will always validate it. You can
+provide it using `--hash`, but as mentioned above, that will turn on hash checking for *all* your
+dependencies. If that is not desirable, use the Cachito-specific `#cachito_hash` URL fragment as
+mentioned in the "Pinning versions" section.
+
+### Build dependencies
+
+Setuptools provides a way to specify build dependencies via the [setup_requires][setuptools-docs]
+keyword argument. It is deprecated in favor of the [PEP-518][pep-518] approach but, for similar
+reasons as mentioned in the sections above, Cachito supports neither. If you have any build-only
+dependencies, you will need to put them in a `requirements-build.txt` file which follows the same
+rules as `requirements.txt`.
```

Comment:
(Note to self): Update this section when we achieve the stretch goal of _Determine build dependencies for user runtime dependencies_